pkg/sip/inbound.go

```diff
@@ -1062,17 +1080,49 @@ func (c *inboundCall) closeWithNoACK() {
 }
 
 func (c *inboundCall) closeWithCancelled() {
-	c.state.DeferUpdate(func(info *livekit.SIPCallInfo) {
-		info.DisconnectReason = livekit.DisconnectReason_CLIENT_INITIATED
-	})
-	c.close(false, CallHangup, "cancelled")
+	var reason ReasonHeader
+	if p := c.closeReason.Load(); p != nil {
+		reason = *p
+	}
+	c.closeWithReason(CallHangup, "cancelled", reason)
 }
 
 func (c *inboundCall) closeWithHangup() {
+	var reason ReasonHeader
+	if p := c.closeReason.Load(); p != nil {
+		reason = *p
+	}
+	c.closeWithReason(CallHangup, "hangup", reason)
+}
+
+func (c *inboundCall) closeWithReason(status CallStatus, reasonName string, reason ReasonHeader) {
 	c.state.DeferUpdate(func(info *livekit.SIPCallInfo) {
 		info.DisconnectReason = livekit.DisconnectReason_CLIENT_INITIATED
+		if info.Error == "" {
+			if !reason.IsZero() {
+				info.Error = reason.String()
```

Comment:
Do we want to populate error for all reasons ?

Reply:
Good point. I'll check for a normal clearing here.